### PR TITLE
ui: fix custom offering cpuspeed during vm import

### LINKED ISSUE #7420: Unable to import instance from VMware to CS
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI
~~~

##### CLOUDSTACK VERSION

~~~
4.18.0.0
~~~

##### CONFIGURATION

~~~
advanced networking, vmware zone
~~~

##### OS / ENVIRONMENT

OracleLinux 8, vSphere Client version 6.7.0.54000

##### SUMMARY

Error "Please enter a valid value.: CPU (in MHz)" when import VM from VMware,

##### STEPS TO REPRODUCE

1. In VMware Vsphere create, clone or restore VM by Veeam. Connect it to existing guest network
2. Find VM in CS "Import-Export instances" section.
3. Select it and push "Import Instance"
4. Fill required fields, choose right compute offering
5. Push "OK" button

##### EXPECTED RESULTS

~~~
VM is imported to CS
~~~

##### ACTUAL RESULTS

~~~
Get error "Please enter a valid value.: CPU (in MHz)"
~~~

image
image

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Fixes #7420

Fixes CPU speed param value when using custom compute offering during VM import

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):

https://user-images.githubusercontent.com/153340/231072144-f360de75-951d-4617-a3b6-a91b348cb0a2.mp4

### CODE REVIEW

**Review by rohityadavcloud — APPROVED** [review]
LGTM, didn't test it though

File: ui/src/views/tools/ImportUnmanagedInstance.vue [hunk 1]
```diff
@@ -522,6 +523,9 @@ export default {
       }
       return 'serviceofferingdetails' in this.computeOffering ? this.computeOffering.serviceofferingdetails.maxmemory * 1 : Number.MAX_SAFE_INTEGER
     },
+    getCPUSpeed () {
+      return (this.computeOffering.iscustomized ? this.resource.cpuspeed : 0) || this.computeOffering?.cpuspeed * 1 || 0
```
[the remark this answers is not in the document]
Comment (shwstppr) [reply ui/src/views/tools/ImportUnmanagedInstance.vue]: refactored

**Comment by rohityadavcloud** [comment]
@blueorangutan ui

File: ui/src/views/tools/ImportUnmanagedInstance.vue [hunk 2]
```diff
@@ -522,6 +523,15 @@ export default {
...
       return 'serviceofferingdetails' in this.computeOffering ? this.computeOffering.serviceofferingdetails.maxmemory * 1 : Number.MAX_SAFE_INTEGER
     },
+    getCPUSpeed () {
+      if (!this.computeOffering) {
+        return 0
+      }
+      if (this.computeOffering.cpuspeed) {
+        return this.computeOffering.cpuspeed * 1
+      }
+      return this.resource.cpuspeed * 1 || 0
```
Comment (DaanHoogland) [inline ui/src/views/tools/ImportUnmanagedInstance.vue:533]: is the `|| 0` needed here?

Comment (rohityadavcloud) [reply ui/src/views/tools/ImportUnmanagedInstance.vue:533]: if the NaN is created from previous expression, then NaN || 0 will return 0. I think this LGTM.

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Pull request merged**